test/integration/statefulset/util.go

```diff
@@ -0,0 +1,329 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statefulset
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"k8s.io/api/apps/v1beta1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	typedv1beta1 "k8s.io/client-go/kubernetes/typed/apps/v1beta1"
+	typedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+	//svc "k8s.io/kubernetes/pkg/api/v1/service"
+	"k8s.io/kubernetes/pkg/controller/statefulset"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+const (
+	interval = 100 * time.Millisecond
+	timeout  = 60 * time.Second
+
+	fakeImageName = "fake-name"
+	fakeImage     = "fakeimage"
+)
+
+type statefulsetTester struct {
+	t           *testing.T
+	c           clientset.Interface
+	service     *v1.Service
+	statefulset *v1beta1.StatefulSet
+}
+
+func labelMap() map[string]string {
+	return map[string]string{"foo": "bar"}
+}
+
+// newService returns a service with a fake name for StatefulSet to be created soon
+func newHeadlessService(namespace string) *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "fake-service-name",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "None",
+			Ports: []v1.ServicePort{
+				{Port: 80, Name: "http", Protocol: "TCP"},
+			},
+			Selector: labelMap(),
+		},
+	}
+}
+
+// newSTS returns a StatefulSet with with a fake container image
+func newSTS(name, namespace string, replicas int) *v1beta1.StatefulSet {
+	replicasCopy := int32(replicas)
+	return &v1beta1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "StatefulSet",
+			APIVersion: "apps/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: v1beta1.StatefulSetSpec{
+			PodManagementPolicy: v1beta1.ParallelPodManagement,
+			Replicas:            &replicasCopy,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labelMap(),
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labelMap(),
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "fake-name",
+							Image: "fakeimage",
+							// Unnecessary to pass current tests
+							VolumeMounts: []v1.VolumeMount{
+								{Name: "datadir", MountPath: "/data/"},
+								{Name: "home", MountPath: "/home"},
+							},
+						},
+					},
+					// Unnecessary to pass current tests
```

Comment:
Remove all comments of `Unnecessary to pass current tests`. It is fine to include the spec attributes although they are not needed for passing the test. Later on, if we are testing against the attributes, we can update them in the test itself.